### DISCUSSION TITLE
Support for FileField

### DISCUSSION THREAD

**Discussion by raivil** [primer/view_components#1346] [General] [answered]
Hello,
I'm planning on adding file upload support to my app and I'm wondering what would be a good strategy to support it.

Here's some example code:

```
<%= form_with model: product do |f| %>
  <%= render(ProductForm.new(f)) %>
<% end %>


class ApplicationForm < Primer::Forms::Base
end

class ProductForm < ApplicationForm
  form do |my_form|
    my_form.text_field(
      name: :title,
      label: "Title",
      required: true
    )

    my_form.file_field(
      name: :my_file,
      label: :description
    )
    my_form.submit(name: :submit, label:"Submit, scheme: :primary, mr: 3) do |c|
      c.with_leading_visual_icon(icon: :"check-circle")
    end
end
end
```

To use `file_field` on a Primer::Forms::Base gem would need to be updated to include `FileFieldInput`, a `FileField` (and erb template) and a new method to `module InputMethods` (incomplete code available at https://github.com/raivil/view_components/pull/1/files).

Do you have any other suggestions that would be simpler? 

Thank you!

**Comment by camertron** [comment, marked as answer]
Hey @raivil thanks for asking about this! What you've outlined above is exactly the right way to go. I realize it's a lot. I'm hoping to remove the need to create a separate `Input` class as well as a component class, but right now it's required 😄
